scripts/pipeline/set_environment_variables/set_pact.sh

```
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+set -e
+#account is shared-prod as all env use prod.
+PACT_BROKER_ACCOUNT="679638075911"
+PACT_BROKER_HTTP_AUTH_USER="admin"
+PACT_BROKER_BASE_URL="pact-broker.api.opg.service.justice.gov.uk"
+PACT_API_VERSION="v1"
+
+if [ $(aws --version 2>&1 | grep -c "aws-cli") -ne 0 ]
+then
+  export SECRET_STRING=$(aws sts assume-role \
+  --role-arn "arn:aws:iam::${PACT_BROKER_ACCOUNT}:role/get-pact-secret-development" \
+  --role-session-name AWSCLI-Session | \
+  jq -r '.Credentials.SessionToken + " " + .Credentials.SecretAccessKey + " " + .Credentials.AccessKeyId' 2>/dev/null)
+
+  #local export so they only exist in this stage
+  export AWS_ACCESS_KEY_ID=$(echo "${SECRET_STRING}" | awk -F' ' '{print $3}' 2>/dev/null)
```

just a query, whats the reason for pumping stderr to /dev/null?

---

just read this - https://stackoverflow.com/questions/40714202/what-is-the-meaning-of-2-in-2-dev-null# think explains it. i guess we are not bothered with the errors coming out.

---

This is taken from the other consumer products. I didn't see a reason not to suppress the error messages. I might rewrite in the future to make it fit more stylistically with the rest of our bash, but for now we know it gets the secrets and sets some values. More will come when we actually implement the tests.